### PR TITLE
239 ui upload agreement letter

### PR DESCRIPTION
**Pull request by philipp-bode**

### CODE REVIEW

**Review by hefowe — APPROVED** [review]
LGTM 👌

File: app/views/application_letters/check.html.erb [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
-  <p class='alert alert-info' >
-    <% if @application_deadline_exceeded %>
-      <%= t('.deadline_exceeded') %>
-    <% else %>
-      <%= t('.can_change_until', application_deadline: l(@application_letter.event.application_deadline)) %>
-    <% end %>
-  </p>
 </div>
 
+<%# Be careful: user.requires_agreement_letter_for_event? behaves exactly the way you wouldn't expect it do behave: If the user requires an agreement letter for the event, it returns false... But fuck me if I'm the one to refactor it%>
```
Comment (bjrne) [inline app/views/application_letters/check.html.erb:14]: I'd much rather see the the refactoring than this comment

File: db/schema.rb [hunk 2]
```diff
@@ -77,10 +77,10 @@
     t.integer  "max_participants"
     t.datetime "created_at",                                 null: false
     t.datetime "updated_at",                                 null: false
-    t.integer  "kind",                       default: 0
-    t.boolean  "published"
     t.string   "organizer"
     t.string   "knowledge_level"
+    t.integer  "kind",                       default: 0
+    t.boolean  "published"
```
Comment (bjrne) [inline db/schema.rb:83]: unnecessary changes

File: spec/controllers/agreement_letters_controller_spec.rb [hunk 3]
```diff
@@ -7,14 +7,15 @@
       @user = FactoryGirl.create(:user, role: :pupil)
       @user.profile ||= FactoryGirl.create(:profile)
       @event = FactoryGirl.create(:event)
+      @application = FactoryGirl.create(:application_letter, user: @user, event: @event)
       filepath = Rails.root.join('spec/testfiles/actual.pdf')
       @file = fixture_file_upload(filepath, 'application/pdf')
       sign_in @user
     end
 
     it "redirects to user profile" do
```
Comment (bjrne) [inline spec/controllers/agreement_letters_controller_spec.rb:16]: title is now incorrect

File: spec/models/profile_spec.rb [hunk 4]
```diff
@@ -36,6 +36,18 @@
     expect(profile.age).to eq(@age)
   end
 
+  it "returns correctly whether the profile owner is an adult or not" do
+    profile = FactoryGirl.build(:profile)
+    adult_profile = FactoryGirl.build(:adult_profile)
+
+
+    puts "FUCK #{profile.birth_date}: #{profile.birth_date >= 18.years.ago}"
+    puts "FUCK #{adult_profile.birth_date}: #{adult_profile.birth_date >= 18.years.ago}"
```
Comment (bjrne) [inline spec/models/profile_spec.rb:45]: what the "FUCK" is this?

**Review by bjrne — COMMENTED** [review]
I am not content with the changes of this PR, although it already got merged. Please look into it again @Philipp-Bode @hefowe

File: app/views/application_letters/check.html.erb (hunk 1 above)
Comment (philipp-bode) [reply app/views/application_letters/check.html.erb:14]: Yea, same here, but unfortunately the scope of the user story did not allow any further work invested into it

File: spec/models/profile_spec.rb (hunk 4 above)
Comment (philipp-bode) [reply spec/models/profile_spec.rb:45]: Missed debug output, sorry for that one

File: spec/controllers/agreement_letters_controller_spec.rb (hunk 3 above)
Comment (philipp-bode) [reply spec/controllers/agreement_letters_controller_spec.rb:16]: The description of the test? I see, will fix that.

File: app/views/application_letters/check.html.erb (hunk 1 above)
Comment (bjrne) [reply app/views/application_letters/check.html.erb:14]: @Philipp-Bode Can you open a technical debt issue for this? 🙂
